noir-projects/noir-contracts/contracts/token_with_refunds_contract/src/main.nr

```diff
@@ -473,22 +472,27 @@ contract TokenWithRefunds {
         // function has access to the final transaction fee, which is needed to compute the actual refund amount.
         context.set_public_teardown_function(
             context.this_address(),
-            FunctionSelector::from_signature("complete_refund((Field,Field,bool),(Field,Field,bool))"),
+            FunctionSelector::from_signature("complete_refund((Field,Field,bool),(Field,Field,bool),Field)"),
             [
-            slotted_fee_payer_point.x, slotted_fee_payer_point.y, slotted_fee_payer_point.is_infinite as Field, slotted_user_point.x, slotted_user_point.y, slotted_user_point.is_infinite as Field
+            slotted_fee_payer_point.x, slotted_fee_payer_point.y, slotted_fee_payer_point.is_infinite as Field, slotted_user_point.x, slotted_user_point.y, slotted_user_point.is_infinite as Field, funded_amount
         ]
         );
     }
 
     #[aztec(public)]
     #[aztec(internal)]
-    fn complete_refund(fee_payer_point: Point, user_point: Point) {
+    fn complete_refund(fee_payer_point: Point, user_point: Point, funded_amount: Field) {
         // 1. We get the final note hiding points by calling a `complete_refund` function on the note.
         // We use 1:1 exchange rate between fee juice and token. So using `tx_fee` is enough
         let tx_fee = context.transaction_fee();
-        let (fee_payer_note_hash, user_note_hash) = TokenNote::complete_refund(fee_payer_point, user_point, tx_fee);
 
-        // 2. At last we emit the note hashes.
+        // 2. We check that user funded the fee payer contract with at least the tx fee.
+        assert(!funded_amount.lt(tx_fee), "tx fee is higher than funded amount");
```

Comment:
Can't we do `funded_amout >= tx_fee`? The negated `lt` makes my brain hurt :sweat_smile:

Reply:
Boolean language also makes my brain hurt but doing it as you proposed would require casting to some uint (you can't do >= on fields) which I didn't do because it would add additional constraint on how large the amounts can be (field is ~254 bits but u64 is just 64).